packages/mask/popups/components/TokenPicker/index.tsx

```diff
@@ -63,35 +81,74 @@ export const TokenPicker = memo(function TokenPicker({
     const isSmartPay = !!useWallet()?.owner
     const networks = useNetworks(NetworkPluginID.PLUGIN_EVM, true)
     const filteredNetworks = useMemo(() => {
-        if (isSmartPay) return networks.filter((x) => x.chainId === ChainId.Polygon && !x.isCustomized)
-        return networks
-    }, [networks, isSmartPay])
+        const list = isSmartPay ? networks.filter((x) => x.chainId === ChainId.Polygon && !x.isCustomized) : networks
+        return chains ? list.filter((x) => chains.includes(x.chainId)) : list
+    }, [chains, networks, isSmartPay])
 
     return (
         <Box className={cx(classes.picker, className)} {...rest}>
-            <SelectNetworkSidebar
-                className={classes.sidebar}
-                networks={filteredNetworks}
-                pluginID={NetworkPluginID.PLUGIN_EVM}
-                chainId={sidebarChainId}
-                hideAllButton
-                onChainChange={handleChainChange}
-            />
-            <List className={classes.list} data-hide-scrollbar>
-                {availableAssets.map((asset) => {
-                    const network = filteredNetworks.find((x) => x.chainId === asset.chainId)!
-                    const selected = asset.chainId === chainId && isSameAddress(asset.address, address)
-                    return (
-                        <TokenItem
-                            key={`${asset.chainId}.${asset.address}`}
-                            asset={asset}
-                            network={network}
-                            selected={selected}
-                            onSelect={onSelect}
-                        />
-                    )
-                })}
-            </List>
+            {!lockChainId ?
+                <SelectNetworkSidebar
+                    className={classes.sidebar}
+                    networks={filteredNetworks}
+                    pluginID={NetworkPluginID.PLUGIN_EVM}
+                    chainId={sidebarChainId}
+                    hideAllButton
+                    onChainChange={handleChainChange}
+                />
+            :   null}
+            <FixedSizeList
+                itemCount={availableAssets.length}
+                itemSize={71}
+                height={455}
+                overscanCount={20}
+                itemData={{
+                    tokens: availableAssets,
+                    networks: filteredNetworks,
+                    chainId,
+                    address,
+                    onSelect,
+                }}
+                itemKey={(index, data) => {
+                    const asset = data.tokens[index]
+                    return `${asset.chainId}.${asset.address}`
```

Comment:
```suggestion
                    return `${asset.chainId}.${asset.address.toLowerCase()}`
```

Reply:
`itemKey` is just used as a identifier, besides, `asset.address` is stable, no any runtime modification. I don't think we need to do this.